src/handler.ts

```diff
@@ -117,18 +117,24 @@ export const serializeCookie = (cookies: Context['set']['cookie']) => {
 
 // 	return arr
 // }
-
 const handleStream = async (
 	generator: Generator | AsyncGenerator,
 	set?: Context['set'],
 	request?: Request
 ) => {
-	let init = generator.next()
-	if (init instanceof Promise) init = await init
-
-	if (init.done) {
-		if (set) return mapResponse(init.value, set, request)
-		return mapCompactResponse(init.value, request)
+	let init
```

Comment:
You can remove try catch in here, `handleStream` function is already wrapped in try-catch that handle onError already

Reply:
Currently the app will crash if you throw an error in a generator before yield, i added a test that reproduces this here: https://github.com/remorses/elysia/blob/932951679c89aef6bad08779b6eb7fe94c30335d/test/response/stream.test.ts#L75

Reply:
I fixed this crash removing the try catch when aot is false. In aot mode it still crashes.

the app crashes because the error is throws asynchronously while mapResponse is a not async, i think there are many more of these bugs hidden in the code, the ideal solution would be to always await `mapResonse` in aot too